src/configuration_parser.cpp

```diff
@@ -48,22 +50,39 @@ bool ConfigurationParser::ParseEnvironmentVariables() {
 }
 
 bool ConfigurationParser::ParseArgV(int argc, char* const argv[]) {
-  // TODO: Parse HITL Variables
-  if (argc < 5) {
+  if (argc < 2) {
     std::cout << "This is a JSBSim integration for PX4 SITL/HITL simulations" << std::endl;
-    std::cout << "   Usage: " << argv[0] << "<aircraft_path> <aircraft> <config> <scene> <headless>" << std::endl;
-    std::cout << "       <aircraft_path>: Aircraft directory path which the <aircraft> definition is located e.g. "
-                 "`models/Rascal`"
-              << std::endl;
-    std::cout << "       <aircraft>: Aircraft file to use inside the <aircraft_path> e.g. Rascal110-JSBSim"
-              << std::endl;
+    std::cout << "   Usage: " << argv[0] << "<config> [-s <scene>]" << std::endl;
     std::cout << "       <config>: Simulation config file name under the `configs` directory e.g. rascal" << std::endl;
     std::cout << "       <scene>: Location / scene where the vehicle should be spawned in e.g. LSZH" << std::endl;
     return false;
   }
 
-  // TODO: Switch to getopt
-  _init_script_path = std::string(argv[4]);
+    static const struct option options[] = {
+        {"scene", required_argument, nullptr, 's'},
+    };
+
+    int c;
+    while ((c = getopt_long(argc, argv, ":s:h", options, nullptr)) >= 0) {
+        switch (c) {
+            case 'h': {
+              break;
+            }
+            case 's': {
+              _init_script_path = std::string(optarg);
+              break;
+            }
+            case ':': {
+                // Do nothing if an empty variable has been passed for all options
+                break;
```

Comment:
Why do you ignore that?

Reply:
I find it useful to do this when doing automated deployments rather than handling "-flag" with checking whether the environment variable is not empty. Should I remove this?

Reply:
Most programs will throw an error, and this can lead to incorrect results with positional arguments, e.g. `./app -s $var script`.
So the idea makes sense, but I'd rather not rely on it.

Reply:
Makes sense, done!